### PR TITLE
Update auth documentation

### PR DESCRIPTION
**Pull request by cmouse**

### CODE REVIEW

File: source/configuration_manual/authentication/password_schemes.rst [hunk 1]
```diff
@@ -164,6 +164,10 @@ different schemes for a single user.
 +---------------+------------------------------------------------------------------------+--------------------------------+
 | SCRAM-SHA-256 | Stronger replacement for SCRAM-SHA-1                                   | .. versionadded:: 2.3.10       |
 +---------------+------------------------------------------------------------------------+--------------------------------+
+| GSSAPI        | Generic Security Services Application Program Interface based          |                                |
+|               | authentication that can be used to authenticate in Kerberos-based      |                                |
+|               | environments like Active Directory or FreeIPA.                         |                                |
++---------------+------------------------------------------------------------------------+--------------------------------+
```
Comment (stephanbosch) [inline source/configuration_manual/authentication/password_schemes.rst:170]: GSSAPI is not a password scheme.

File: source/configuration_manual/authentication/authentication_mechanisms.rst [hunk 2]
```diff
@@ -46,49 +46,48 @@ cleartext password.
...
+| OTP                                                                          | One time password mechanisms.                                            |                             |
++------------------------------------------------------------------------------+--------------------------------------------------------------------------+-----------------------------+
+| EXTERNAL                                                                     | EXTERNAL SASL mechanism.                                                 |                             |
++------------------------------------------------------------------------------+--------------------------------------------------------------------------+-----------------------------+
+| OAUTHBEARER                                                                  | OAuth2 bearer authentication :rfc:`7628`.                                | .. versionadded:: 2.2.29    |
+|                                                                              | See :ref:`authentication-oauth2`.                                        |                             |
```
Comment (stephanbosch) [inline source/configuration_manual/authentication/authentication_mechanisms.rst:86]: Place of reference is inconsistent with other mechanisms above (those have it in the mechanism column if I understand the syntax correctly).

File: source/configuration_manual/authentication/gssapi.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+This should enable GSSAPI support for Dovecot.
+
+Testing
+=======
+
+You need an GSSAPI capable client, such as Thunderbird, running on a fully working Kerberos workstation. You can choose GSSAPI as your authentication mechanism
+and you should be able to log in.
+
+If necessary, you can use auth_username_format to normalize usernames.
```
Comment (stephanbosch) [inline source/configuration_manual/authentication/gssapi.rst:73]: Cross-realm authentication is not documented.

This also involves the following passdb field (depending on the environment, apparently not for Solaris.. eww):

https://github.com/dovecot/core/blob/main/src/auth/mech-gssapi.c#L412

Comment (cmouse) [reply source/configuration_manual/authentication/gssapi.rst:73]: Spent half day googling for that only to find it in the source code ... :)

File: source/configuration_manual/authentication/password_schemes.rst [hunk 4]
```diff
@@ -185,19 +185,31 @@ Strong schemes and mechanism-specific schemes are listed above.
...
 * **PLAIN-MD5**: MD5 sum of the password stored in hex.
 * **LDAP-MD5**: MD5 sum of the password stored in base64.
 * **SMD5**: Salted MD5 sum of the password stored in base64.
+* **HMAC-MD5**: Alias CRAM-MD5.
 
 SHA based schemes (also see below for libc's SHA* support):
 ***********************************************************
 
 * **SHA**: SHA1 sum of the password stored in base64.
+* **SHA1**: SHA1 sum of the password stored in base64.
```
Comment (sirainen) [inline source/configuration_manual/authentication/password_schemes.rst:212]: `Alias for SHA`? Like you wrote for MD5 ones.

**Review by sirainen — CHANGES_REQUESTED** [review]

**Review by sirainen — APPROVED** [review]

**Pull request merged**